rdt/hyper_transformer.py

```diff
@@ -320,6 +324,40 @@ def get_transformer_tree_yaml(self):
 
         return yaml.safe_dump(dict(modified_tree))
 
+    def detect_initial_config(self, data):
+        """Print the configuration of the data.
+
+        This method detects the ``sdtype`` and transformer of each field in the data
+        and then prints them as a json object.
+
+        NOTE: This method partially resets the state of the ``HyperTransformer``.
+        Previously set ``sdtypes`` or transformers will be lost.
+
+        Args:
+            data (pd.DataFrame):
+                Data which will have its configuration detected.
+        """
+        # Reset the state of the HyperTransformer
+        self.field_data_types = {}
+        self.field_transformers = {}
+
+        # Set the sdtypes and transformers of all fields to their defaults
+        for field in data:
+            self._set_field_data_type(data, field)
+            field_sdtype = self.field_data_types[field]
+            self.field_transformers[field] = get_default_transformer(field_sdtype)
+
+        print('Detecting a new config from the data ... SUCCESS')  # noqa: T001
+        print('Setting the new config ... SUCCESS')  # noqa: T001
```

Comment:
Is there a way to add this warning for the whole method instead of on each line where we print?

Reply:
I'm a bit skeptical about the whole printing but here is my proposal, I know I'm iterating twice on the data but I don't see a better way to do it:
```python
        # Set the sdtypes and transformers of all fields to their defaults
        print('Detecting a new config from the data ... ', end='')  # noqa: T001
        try:
            for field in data:
                self._set_field_data_type(data, field)
                field_sdtype = self.field_data_types[field]

            print('SUCCESS')  # noqa: T001
        except Exception e:
            print('ERROR')  # noqa: T001

        print('Setting the new config ... ', end='')  # noqa: T001
        try:
            for field in data:
                self.field_transformers[field] = get_default_transformer(field_sdtype)

             print('SUCCESS')  # noqa: T001
        except Exception e:
            print('ERROR')  # noqa: T001
```

This way we can print `ERROR` or `SUCCESS` and actually not just 'straight forward'  print a message. @npatki  @amontanez24 your thoughts on this ?

Reply:
I'm not sure. If it exists I wasn't able to find it.